rust/src/http2/parser.rs

```diff
@@ -0,0 +1,138 @@
+/* Copyright (C) 2020 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use nom::number::streaming::be_u8;
+use std::fmt;
+
+#[repr(u8)]
+#[derive(Clone, Copy, PartialEq, FromPrimitive, Debug)]
+pub enum HTTP2FrameType {
+    DATA = 0,
+    HEADERS = 1,
+    PRIORITY = 2,
+    RSTSTREAM = 3,
+    SETTINGS = 4,
+    PUSHPROMISE = 5,
+    PING = 6,
+    GOAWAY = 7,
+    WINDOWUPDATE = 8,
+    CONTINUATION = 9,
+}
+
+impl fmt::Display for HTTP2FrameType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+impl std::str::FromStr for HTTP2FrameType {
+    type Err = String;
+
+    //TODO more automatic
```

Comment:
@jasonish any tips here ?

Reply:
Not really, this type of thing is pretty common.

Reply:
Ok, and what about the next line :

This works
```
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s {
```

And I would like this

```
    fn from_str(s: &str) -> Result<Self, Self::Err> {
        match s.to_uppercase() {
```

which does not work like that